### DISCUSSION TITLE
Websocket connection fails with 1011 - Expected OPEN Websocket state, but got 2

### DISCUSSION THREAD

**Discussion by hasinthaK** [speechly/speechly#228] [Q&A] [answered]
I've integrated Speechly polyfill with an Angular application, implemented `onresult` & `onerror` callbacks after checking `browserSupport`, 
I can confirm that the initial connection to the Speechly API is okay since I had not added the App Id at first. After adding it all errors have gone away. Except, when I start recording audio, it errors out with below error, it seems like the library or the browser is having some trouble keeping the websocket connection alive for data sending & receiving.

Any idea on what's happening?

Below is the handler method I'm using for start & stop recording, 

`
   handleSpeechToTextOnDetailedView() {
      if (this.isSpeechToTextOn) {
        this.resetSpeechToText();
        return;
      }
    const SpeechlySpeechRecognition = createSpeechlySpeechRecognition(environment.speechlyAppId);
    if (SpeechlySpeechRecognition.hasBrowserSupport) {
      this.isSpeechToTextOn = true;
      this.speechRecognitionObj = new SpeechlySpeechRecognition();

      // Get transcripts while the user is speaking, not just when they've finished
      this.speechRecognitionObj.continuous = true;
      this.speechRecognitionObj.interimResults = true;

      this.speechRecognitionObj.onresult = ({ results }: any) => {
        console.debug({ results });
      };

      this.speechRecognitionObj.onerror = (event: any) => {
        if (event === MicrophoneNotAllowedError) {
          this.toastr.error('You denied access to your microhpne.');
        } else {
          this.toastr.error('Unknown error occurred while trying to process speech to text.');
        }
        this.resetSpeechToText();
      };

      this.speechRecognitionObj.start();
    } else {
      this.toastr.warning('Seems like your browser doesn\'t support speech to text.');
    }
  }

  private resetSpeechToText() {
    this.isSpeechToTextOn = false;
    this.speechRecognitionObj.stop();
    this.speechRecognitionObj = undefined;
  }
`

![image](https://user-images.githubusercontent.com/52355600/195570083-dd5ef090-65c5-4750-86ad-39e30bad78d8.png)

Appreciate any help or elaboration on this,

BTW great polyfill with a good trial usage!

**Comment by arzga** [comment, marked as answer]
Hi,

Unfortunately browser-client/react-client in npm released 2 days ago was out-of-sync with our backend. The backend should now be updated to support the changes.

Could you retry if it works now?

**Comment by RyanJeon** [comment]
@arzga Hello, I am running into the same error for the past 2 weeks. In my case, the SDK keeps working for a few minutes and it will suddenly trigger this error. Has anyone encountered this problem before?
